### DISCUSSION TITLE
Cheapest slots - too many hours

### DISCUSSION THREAD

**Discussion by dougle03** [BottlecapDave/HomeAssistant-OctopusEnergy#1077] [Q&A]
Hi all,
Below are my cheapest sensor settings and the results from last night.
What am I doing wrong? 
These are my settings: Clearly asking for the cheapest 2.5 hours only.
![image](https://github.com/user-attachments/assets/cdbf4f0c-7450-40e1-a313-e4873bdeea10)
![image](https://github.com/user-attachments/assets/a83a0461-5fc2-4aca-9710-0cf736bf996f)

And this is the output from the sensor:
![image](https://github.com/user-attachments/assets/f098b2b9-622f-49c3-90c5-a789a4a1c421)
As you can see the sensor was on for 5 hours!

Am I doing something wrong?
Thanks

**Comment by BottlecapDave** [comment]
Hello. You want to turn off rolling target, as this will re-evaluate once the current period has passed and find a new period if time permits.

**Reply by dougle03** [reply]
Hi,
Thanks, I'll try that tonight; although as luck would have it tomorrow's forecast is sunny, but i'll be able to look at the sensor history.
Thanks again, P

**Reply by dougle03** [reply]
It worked... yey.
Whilst I'm on, is there a way to automate the 'hours required' value?
I automate with Node Red so being able to on-the-fly adjust the overall duration of the target duration would be really useful.
Thanks.

**Reply by BottlecapDave** [reply]
You'll want to look at https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/services/#octopus_energyupdate_target_config

**Reply by dougle03** [reply]
Much appreciated; now I just need to work out how to call the `input_number.octopus_energy_target_hours` 
This is what I have, it's not erroring, but the target hours in the integration are not changing. Is there a delay between the call and the integration showing the updated value?
![image](https://github.com/user-attachments/assets/e60c2e41-8ecc-4541-984b-675abaab6f26)

Thanks again for your help.

**Reply by BottlecapDave** [reply]
input_number was a custom helper sensor which allows the value to be exposed and editable as part of a dashboard. If you don't need this, you can just call the service directly.

The target rate should be updated straight away, but I'm not familiar with node red so don't know if there's a delay there. Hour needs to be in increments of 0.5. if called directly in HA you should get an error if the request is invalid.

**Reply by dougle03** [reply]
Hi Dave,
Thanks for responding.
I'm looking at my settings and can't see any delay. Note that I use NodeRed for all my automations (Long story), usually when a call service node fails it's errors with a message.
What's odd is that this is not erroring, but the number of hours does not change (When I go into the config of the Octopus integration). Really can't find any reason for it to not be working...
I'll keep plugging away, but let me know if you can think of anything. Thanks, P

**Reply by BottlecapDave** [reply]
Can you change the value via the developer tools and see it reflected in the sensor?
